sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/LN0Adapter.java

```diff
@@ -177,42 +182,30 @@ private ResumedDataTemplate getResumedDataTemplate(String doName, DaTypeName daT
         final String ldInst = getParentAdapter().getInst();
-        DaTypeName daTypeNameBeh = new DaTypeName();
-        daTypeNameBeh.setName(STVAL);
-        daTypeNameBeh.setBType(TPredefinedBasicTypeEnum.ENUM);
-        daTypeNameBeh.setFc(TFCEnum.ST);
-        ResumedDataTemplate daiBehFilter = getResumedDataTemplate(BEHAVIOUR_DO_NAME, daTypeNameBeh);
+        ResumedDataTemplate daiBehFilter = getDataTypeFilter(BEHAVIOUR_DO_TYPE_NAME, BEHAVIOUR_DA_TYPE_NAME);
         List<ResumedDataTemplate> daiBehList = getDAI(daiBehFilter, false);
         if (daiBehList.isEmpty()) {
-            errors.add(buildErrorDescriptionMessage("The LDevice doesn't have a DO @name='Beh' OR its associated DA@fc='ST' AND DA@name='stVal'"));
-            return errors;
+            return Optional.of(buildFatalReportItem("The LDevice doesn't have a DO @name='Beh' OR its associated DA@fc='ST' AND DA@name='stVal'"));
         }
         Set<String> enumValues = getEnumValues(daiBehList.get(0).getDaName().getType());
         List<TCompasLDevice> compasLDevicePrivateList = PrivateService.getCompasPrivates(getParentAdapter().getCurrentElem(), TCompasLDevice.class);
         if (compasLDevicePrivateList.isEmpty()) {
-            errors.add(buildErrorDescriptionMessage("The LDevice doesn't have a Private compas:LDevice."));
-            return errors;
+            return Optional.of(buildFatalReportItem("The LDevice doesn't have a Private compas:LDevice."));
         }
         if (!compasLDevicePrivateList.get(0).isSetLDeviceStatus()) {
-            errors.add(buildErrorDescriptionMessage("The Private compas:LDevice doesn't have the attribute 'LDeviceStatus'"));
-            return errors;
+            return Optional.of(buildFatalReportItem("The Private compas:LDevice doesn't have the attribute 'LDeviceStatus'"));
         }
         TCompasLDeviceStatus compasLDeviceStatus = compasLDevicePrivateList.get(0).getLDeviceStatus();
-        DaTypeName daTypeNameMod = new DaTypeName();
-        daTypeNameMod.setName(STVAL);
-        ResumedDataTemplate daiModFilter = getResumedDataTemplate(MOD_DO_NAME, daTypeNameMod);
-        List<ResumedDataTemplate> daiModList = getDAI(daiModFilter, false);
-        if (daiModList.isEmpty()) {
-            errors.add(buildErrorDescriptionMessage("The LDevice doesn't have a DO @name='Mod'"));
-            return errors;
+        Optional<ResumedDataTemplate> optionalModStVal = getDaiModStVal();
+        if (optionalModStVal.isEmpty()) {
+            return Optional.of(buildFatalReportItem("The LDevice doesn't have a DO @name='Mod'"));
         }
-        ResumedDataTemplate newDaModToSetInLN0 = daiModList.get(0);
-        String initialValue = newDaModToSetInLN0.getDaName().getDaiValues().isEmpty() ? "" : newDaModToSetInLN0.getDaName().getDaiValues().values().toArray()[0].toString();
+        ResumedDataTemplate newDaModToSetInLN0 = optionalModStVal.get();
+        String initialValue = newDaModToSetInLN0.findFirstValue().orElse("");
```

Comment:
why "" ? instead of null?

Reply:
It is a refacto of the line : 
```java
String initialValue = newDaModToSetInLN0.getDaName().getDaiValues().isEmpty() ? "" : newDaModToSetInLN0.getDaName().getDaiValues().values().toArray()[0].toString();
```
I created the method `ResumedDataTemplate.findFirstValue` in this issue, and I took the opportunity to refactor this line. But I don't have the business details.